src/Validators/PatientValidator.php

```diff
@@ -26,12 +26,12 @@ class PatientValidator extends BaseValidator
      */
     public function isExistingPid($pid)
     {
-        $rtn = sqlQuery(
-            "SELECT pid FROM patient_data WHERE pid = ?",
+        $result = sqlQuery(
+            "SELECT pid AS pid FROM patient_data WHERE pid = ?",
             array($pid)
-        )['pid'];
+        );
 
-        $isExisting = $rtn == null ? false : true;
+        $isExisting = !($result["pid"] == null);
```

Comment:
maybe just do:
```
$isExisting = !empty($result["pid"]);
```
(shouldn't need the changes above this)
If need multiple php environements locally to test quickly, then there is the insane docker development environment here :)
https://github.com/openemr/openemr/tree/master/contrib/util/docker#insane-development-docker-environment

Reply:
Hi @dixonwhitmire, @bradymiller 

From a design prospective  the validation should fail before the execution of the query.

you can improve the design  with a typeint or a guard clause.
```bash
│ /usr/share/nginx/html/openemr/tests/Tests/Services/PatientServiceTest.php:82
```
See the above test execute an update with ""not-a-pid"" as a pid

option 1: use int type in the signature
```php
public function isExistingPid(int $pid)
```

option 2: use guard clause
```php
public function isExistingPid($pid)
{
  //guard clause here 
  // fail with message if pid is a string

}
```

Reply:
@bradymiller - I was able to update the container to php 7.4 and verified that this works
```
$pidValue = $result["pid"] ?? 0;
return $pidValue > 0;
```
The Travis CI builds target PHP 7.4, so I'm assuming this will be ok (apparently "??" is specific to PHP 7). If I need to tweak it just let me know!

Reply:
Hi @pMononoke!

First of all, thank you for taking the time to review this PR.

I would be happy to update the validation function. From what I've seen of the project, I don't know if types are used in method/function signatures, so I would opt for option 2.

Is something like this what you have in mind?
```
if !is_int($pid) {
   return false;
```
Thanks!

Reply:
@dixonwhitmire , solution looks good :)

Reply:
@dixonwhitmire 
yes! 
 I'm a little bit paranoic :-)   for my POV even negative integer are invalid value for $pid variable.